### DISCUSSION TITLE
Elastalert emails not sending

### DISCUSSION THREAD

**Discussion by Chris-GNS** [Security-Onion-Solutions/securityonion#6882] [General]
Elastalert fails to email alerts.

It was working but I updated to 2.3.91 to fix log4j vulnerability and the emails stopped.
The so-elastalert-test completes successfully but no email received.

The elastalert log has this error:
elastalert Error while running alert email: Error connecting to SMTP host: [SSL: DH_KEY_TOO_SMALL] dh key too small (_ssl.c:1129)

Then I added TLS support to the configuration and get a new error:
elastalert Error while running alert email: Error connecting to SMTP host: [SSL: UNSUPPORTED_PROTOCOL] unsupported protocol (_ssl.c:1129)

***Revelant info:
Exact version = 2.3.91
Install source = Security Onion ISO image

Install type = Standalone with a Heavy Node

Does so-status show all services running? = services running

Do you get any failures when you run salt-call state.highstate from the node? = None

**Comment by Chris-GNS** [comment]
Still nothing.

**Comment by weslambert** [comment]
`so-elastalert-test` won't send an email on it's own, unless you tell it to do so (`a` option):

```
[root@securityonion]# so-elastalert-test -h

Test Elastalert Rule
  Options:
  -h                This message
  -a                Trigger real alerts instead of the debug alert
  -l <path_to_file> Write results to specified log file
  -o '<options>'    Specify Elastalert options ( Ex. --schema-only , --count-only, --days N )
  -r <rule_name>    Specify filename of rule to test (must exist in /opt/so/rules/elastalert; do not include path)
```

- Are you attempting to connect to an internal mail relay, or something like Gmail? 
- What is the TLS version you are trying to use? 
- Could you provide an example of your configuration?
